### DISCUSSION TITLE
Using pdpipe pipelines with GridSearchCV for parameter tuning

### DISCUSSION THREAD

**Discussion by stefansimik** [pdpipe/pdpipe#81] [Q&A] [answered]
Hi, please add to documentation
simple example for common scenario in ML models,
how to use pdpipe with GridSearchCV.

It is not clear, how to do it / of if expected scenario at all.

This basic simple will help people to use pdpipe for all 
ML models and parameter-tuning techniques.

Best
Stefan

**Comment by shaypal5** [comment]
Good question. I'm working on it for the last couple of days.

I am not yet sure how easy it is to implement, as I think `sklearn` casts input dataframes to `numpy.ndarray` objects, but we'll see.

You can take a look at this branch if you want to:
https://github.com/pdpipe/pdpipe/tree/custpipebase

**Comment by shaypal5** [comment]
I made it work! :)

**Comment by shaypal5** [comment, marked as answer]
Released in `v0.0.66`.

I've added a tutorial page on the doc site, walking users through how to perform GridSearchCV with pdpipe pipelines:
https://pdpipe.readthedocs.io/en/latest/tutorials/skintegrate/#grid-search-cross-validation-with-pipeline-object-models

I've also written a tutorial notebook just for you (and also for future users, but mainly for you❤️):
https://github.com/pdpipe/pdpipe/blob/master/notebooks/pipeline_and_model.ipynb

Let me know about any problems you have, and feel free to open new issues for bugs, missing parts and other features you'd like to have! :)

**Comment by stefansimik** [comment]
Excellent, thank you very much ❤️ 
I will test in my current project 👍

**Comment by shaypal5** [comment]
Great! Let me know how it goes.
